src/MIDebugEngine/AD7.Impl/AD7Disassembly.cs

```diff
@@ -134,10 +134,16 @@ public int Seek(enum_SEEK_START dwSeekStart, IDebugCodeContext2 pCodeContext, ul
             if (iInstructions != 0)
             {
                 IEnumerable<DisasmInstruction> instructions = null;
-                _engine.DebuggedProcess.WorkerThread.RunOperation(async () =>
+                // When failing to read all instructions try looking for smaller and smaller numbers of instructions
+                // so that fetching disassembly works near the start and end of valid address spaces
+                while (instructions == null && (iInstructions > 1 || -iInstructions > 1)) 
                 {
-                    instructions = await _engine.DebuggedProcess.Disassembly.FetchInstructions(_addr, (int)iInstructions);
-                });
+                    _engine.DebuggedProcess.WorkerThread.RunOperation(async () =>
+                    {
+                        instructions = await _engine.DebuggedProcess.Disassembly.FetchInstructions(_addr, (int)iInstructions);
+                    });
+                    iInstructions = iInstructions/2;
```

Comment:
Two questions: 
1. Suppose there are 19 valid instructions before the page offset and we ask for 20 initially. Now we will only wind up only asking for 10. Isn't it a problem that we didn't get those extra 9 valid instructions?
2. For the invalid instructions, don't we want to fill the disassembly window with '??' lines so that it will still scroll. If so, shouldn't we take the difference between what was asked for and the number of valid instructions and seek 1-byte for each invalid?